<launch>
	
	<node pkg="dji_sdk" type="dji_sdk_node" name="dji_sdk" output="screen">
		
		<param name="serial_name" type="string" value="/dev/ttyUSB0"/>
		<param name="baud_rate" type="int" value="230400"/>
		<param name="app_id" type="int" value="1033821"/>
		<param name="app_version" type="int" value="2"/>
		<param name="app_bundle_id" type="string" value="Not required in this SDK version"/>
		<param name="enc_key" type="string" value="65e9bd441ca08d0bc3e933b009840acc460f4809555ec62fcea53a540828d5ea"/>
		<param name="groundstation_enable" type="int" value="1"/>
	</node>

 <rosparam command="load" file="$(find dji_sdk_demo)/config/ros.yaml"/>

<node pkg="dji_sdk_demo" type="dji_sdk_client" name="dji_sdk_client" output="screen">
        <param name="imgWidth" value="640"/>
        <param name="imgHeight" value="480"/>

        <param name="DEBUG_svm" value="false"/>
        <param name="DEBUG_yolo" value="false"/>

        <param name="targetNumber" value="1"/>
        <param name = "initYaw" value="116"/>
        <param name = "offset_kp" value="8"/>
        <param name="right_limit" value="5"/>
        <param name="left_limit" value="-5"/>
        <param name="normalHeightV" value="0.3"/>

	<param name="landMissionH" value="1.2"/>
	<param name="acrossMissionH" value="1.0"/>
        <param name="QRCodeMissionH" value="1.5"/>

        


	</node>
    <arg name="markerSize"      default="0.05"/>    <!-- in m -->
    <arg name="side"             default="left"/>
    <arg name="ref_frame"       default="base"/>  <!-- leave empty and the pose will be published wrt param parent_name -->


    <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher" output="screen">
        <remap from="/camera_info" to="/cameras/$(arg side)_hand_camera/camera_info" />
        <remap from="/image" to="/cameras/$(arg side)_hand_camera/image" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <param name="camera_frame"       value="$(arg side)_hand_camera"/>
<param name="use_camera_info"       value="false"/>

    </node>




</launch>
